services/motion/server.go

```diff
@@ -60,6 +60,39 @@ func (server *serviceServer) MoveOnMap(ctx context.Context, req *pb.MoveOnMapReq
 	return &pb.MoveOnMapResponse{Success: success}, err
 }
 
+func (server *serviceServer) MoveOnMapNew(ctx context.Context, req *pb.MoveOnMapNewRequest) (*pb.MoveOnMapNewResponse, error) {
+	svc, err := server.coll.Resource(req.Name)
+	if err != nil {
+		return nil, err
+	}
+	if req == nil {
+		return nil, errors.New("received nil *pb.MoveOnMapNewRequest")
+	}
+	if req.Destination == nil {
+		return nil, errors.New("must provide a destination")
+	}
+	protoComponentName := req.GetComponentName()
+	if protoComponentName == nil {
+		return nil, errors.New("received nil *commonpb.ResourceName for component name")
+	}
+	protoSlamServiceName := req.GetSlamServiceName()
+	if protoSlamServiceName == nil {
+		return nil, errors.New("received nil *commonpb.ResourceName for SlamService name")
+	}
```

Comment:
I recommend moving these validations into a private fromProto function.

Reply:
Agree with this.  Ideally this whole PR would just be adding parallel code so that it looks as close to MoveOnGlobeNew as possible